src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -800,8 +800,7 @@ JvmtiEnvBase::get_vthread_state(oop thread_oop, JavaThread* java_thread) {
     state = (jint)java_lang_VirtualThread::map_state_to_thread_status(vt_state);
   }
   if (ext_suspended && ((state & JVMTI_THREAD_STATE_ALIVE) != 0)) {
-    state &= ~java_lang_VirtualThread::RUNNING;
-    state |= JVMTI_THREAD_STATE_ALIVE | JVMTI_THREAD_STATE_RUNNABLE | JVMTI_THREAD_STATE_SUSPENDED;
+    state |= JVMTI_THREAD_STATE_SUSPENDED;
   }
```

Comment:
One question unrelated to this bug and your fix.
I wonder if any check and handling is needed for the case:
  `if (ext_suspended && ((state & JVMTI_THREAD_STATE_ALIVE) == 0))`
Not sure this condition is even possible. But do we need to add an assert here?

Reply:
AFAIU it's possible in the case when we have terminated VT and JvmtiVTSuspender is requested to suspend all virtual threads

Reply:
So there is a window in which is a VT is marked as terminated yet is still visible for actions like this? For regular threads we would always have filtered out thread in the process of exiting. Seeing terminated threads seems potentially problematic but perhaps all the VT code is prepared to handle this.

Reply:
I hope there is no such window in GetThreadState() case, but get_vthread_state method is also called from MultipleStackTracesCollector::fill_frames and there is a comment there:
// Note that either or both of thr and thread_oop
// may be null if the thread is new or has exited.

I keep this check for safety (though fill_frames does not care about suspend bit)